### PR TITLE
Allow to pass nil user and toggle feature based on general state

### PR DESCRIPTION
**Pull request by timurvafin**

### CODE REVIEW

File: lib/ffeature/feature.rb [hunk 1]
```diff
@@ -7,11 +7,14 @@ def initialize(feature)
...
+    def enabled?(user = nil)
       return true if FFeature.dev_mode
-      return false unless flipper_user?(user)
 
-      FFeature.flipper[feature].enabled?(user.to_model)
+      if flipper_user?(user)
+        FFeature.flipper[feature].enabled?(user.to_model)
+      else
+        FFeature.flipper[feature].enabled?
+      end
```
Comment (maxinspace) [inline lib/ffeature/feature.rb:17]: tell don't ask, return clause is actually preferable, imho.
```
def enabled?(user = nil)
  return true if FFeature.devmode
  return FFeature.flipper[feature].enabled?(user.to_model) if flipper_user?(user)

  FFeature.flipper[feature].enabled?
end
```

Comment (timurvafin) [reply lib/ffeature/feature.rb:17]: Too late, my friend!

But in this particular case
1. if-else clause shows the difference between checking feature state based on user or not
2. return based is too long to read

Comment (dmitrytrager) [reply lib/ffeature/feature.rb:17]: You have to pass 'user' to some new method/object to get TDA here

File: spec/ffeature_spec.rb [hunk 2]
```diff
@@ -29,19 +29,41 @@
...
-    expect(described_class.enabled?(:archived, user)).to be_falsey
+    it "does not toggle feature for other users" do
+      expect(described_class.enabled?(:archived, user)).to be_falsey
+    end
+
+    it "does not toggle feature for nil" do
+      expect(described_class.enabled?(:archived, nil)).to be_falsey
+      expect(described_class.enabled?(:archived)).to be_falsey
+    end
   end
```
Comment (dmitrytrager) [inline spec/ffeature_spec.rb:50]: Is disabled

**Pull request merged**